src/NovelRT/Graphics/Vulkan/VulkanGraphicsDevice.cpp

```diff
@@ -345,10 +351,14 @@ namespace NovelRT::Graphics::Vulkan
             throw std::runtime_error("Failed to acquire next VkImage! Reason: " +
                                      std::to_string(acquireNextImageResult));
         }
-        _contextIndex = contextIndex;
 
-        presentCompletionGraphicsFence->Wait();
-        presentCompletionGraphicsFence->Reset();
+        if (!_isAttachedToResizeEvent)
```

Comment:
Why is this needed?

Reply:
It's so it can attach on first run. It's either here or on construction, which means making a ton of stuff non-lazy, which means a descent into C++ madness IIRC.

Reply:
I don't remember the specifics but I remember it being problematic in some fashion with some lazies.